server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1579,8 +1579,14 @@ public Tuple<GatedCloseable<SegmentInfos>, ReplicationCheckpoint> getLatestSegme
         if (indexSettings.isSegRepEnabled() == false) {
             return null;
         }
+
+        Tuple<GatedCloseable<SegmentInfos>, ReplicationCheckpoint> nullSegmentInfosEmptyCheckpoint = new Tuple<>(
+            new GatedCloseable<>(null, () -> {}),
+            ReplicationCheckpoint.empty(shardId, getDefaultCodecName())
+        );
+
         if (getEngineOrNull() == null) {
-            return new Tuple<>(new GatedCloseable<>(null, () -> {}), ReplicationCheckpoint.empty(shardId, getDefaultCodecName()));
+            return nullSegmentInfosEmptyCheckpoint;
         }
         // do not close the snapshot - caller will close it.
         final GatedCloseable<SegmentInfos> snapshot = getSegmentInfosSnapshot();
```

Comment:
`getSegmentInfosSnapshot()` also invokes getEngine() and can throw AlreadyClosedException that will not get caught here.  I think we should also catch it and remove the redundant null check on `getEngineOrNull` above.

Reply:
Or have getSegmentInfosSnapshot wrap a null in gatedCloseable.